### ISSUE TITLE
[feature]: enable / disable all buttons to quickly add patterns to playlist

### ISSUE THREAD

**Issue by caleyg** [simap/Firestorm#45]
A feature that would be useful is enabling and disabling all UI buttons to quickly add all the available patterns in the UI instead of only being able to click them each one by one. 

I have been tinkering with different approaches while keeping the original implementation as much as possible. If I am successful, I'll push up a PR with this implemented.

**Comment by caleyg** [comment]
<img width="1533" alt="Screen Shot 2023-04-18 at 10 11 29 AM" src="https://user-images.githubusercontent.com/4196377/232821794-f1acf51b-522d-474b-b609-85a638ad012e.png">


I have the handlers in place, but the `enableAllPatterns` handler takes some time to process behind the scenes. I would love to present the spinner we're using in the CSS, but react is bringing me great pain in the way it batch updates `this.state` until after all processing is complete to avoid any extra re-renders, which can be resource heavy in bigger apps. 

As a result, I also noticed that in the `handleClone` dialog, a similar loader is in place there... and as a result, the behavior I see with my implemented loader is also reproducible when trying to clone patterns from one device to another. 

In other words, the loader never shows because we set `this.setState({ isCloning: true})`, or `this.setState({ isProcessing: true })` in the same function (`handler`) invoked by their respective buttons `onClick` attribute, which later is set to `this.setState({ isCloning: false})`, or `this.setState({ isProcessing: false })`...

I'll keep banging on it to get the loader to show while the `Enable All` button is processing the patterns into the backend (sqlite.db)

**Comment by caleyg** [comment]
<img width="1616" alt="Screen Shot 2023-04-18 at 12 32 00 PM" src="https://user-images.githubusercontent.com/4196377/232858534-14b284aa-210f-44eb-9b3c-9ad9f8bdb65e.png">
whew got it, react lifecycles break my soul sometimes lol

**Comment by caleyg** [comment]
included in this PR https://github.com/simap/Firestorm/pull/44
